src/parser.rs

```diff
@@ -363,6 +363,36 @@ impl<'a> Parser<'a> {
         Ok(expr)
     }
 
+    pub fn parse_interval_expr(&mut self) -> Result<Expr, ParserError> {
+        let precedence = 0;
+        let mut expr = self.parse_prefix()?;
+
+        loop {
+            let next_precedence = self.get_next_interval_precedence()?;
+
+            if precedence >= next_precedence {
+                break;
+            }
+
+            expr = self.parse_infix(expr, next_precedence)?;
+        }
+
+        Ok(expr)
+    }
+
+    /// Get the precedence of the next token
+    /// With AND, OR, and XOR
+    pub fn get_next_interval_precedence(&self) -> Result<u8, ParserError> {
+        let token = self.peek_token();
+
+        match token {
+            Token::Word(w) if w.keyword == Keyword::AND => Ok(0),
+            Token::Word(w) if w.keyword == Keyword::OR => Ok(0),
+            Token::Word(w) if w.keyword == Keyword::XOR => Ok(0),
```

Comment:
why XOR here? 

Also OR and XOR do not appear to be covered in the tests

Reply:
I mostly just included XOR since I figured precedence rules in this case would be similar to AND and OR, and they're also lumped together at https://github.com/sqlparser-rs/sqlparser-rs/blob/4b1dc1abf7138cb1576707c194ec89ad89049d27/src/parser.rs#L1635.

Since the AND test looks ok, I can add OR and XOR tests as well.